core/src/test/java/com/predic8/membrane/core/interceptor/oauth2/client/BrowserMock.java

```diff
@@ -168,9 +168,12 @@ public class BrowserMock implements Function<Exchange, Exchange> {
     private static void addCookiesToExchange(Exchange exc, Map<String, String> cookies) {
         if (cookies == null)
             return;
-        exc.getRequest().getHeader().addAll(cookies.entrySet().stream()
-                .map(BrowserMock::createCookieHeaderField)
-                .toList());
+        Header header = exc.getRequest().getHeader();
+        synchronized (header) {
```

Comment:
TL/DR: Please remove the `synchronized(header)`.

The header here is also used only by one thread. (Accessing an `Exchange` or one of its sub-objects from multiple threads in parallel is always not safe. And this is not done here: Each thread has one `Exchange` for itself.)

If something needed protection, it was `browserMock.cookie`: This is actually read/written to by multiple threads.